designsystem/src/main/java/com/b1nd/dodam/designsystem/component/Card.kt

```diff
@@ -0,0 +1,260 @@
+            AskStatus.ALLOWED -> MaterialTheme.colorScheme.primary to "승인됨"
+            AskStatus.PENDING -> MaterialTheme.colorScheme.tertiary to "대기중"
+            AskStatus.REJECTED -> MaterialTheme.colorScheme.error to "거절됨"
+        }
+        Row(verticalAlignment = Alignment.CenterVertically) {
+            Box(
+                modifier = Modifier
+                    .clip(shape = RoundedCornerShape(100))
+                    .background(statusColor)
+                    .padding(
+                        vertical = 4.dp,
+                        horizontal = 12.dp,
+                    ),
+            ) {
+                Text(
+                    text = statusText,
+                    style = MaterialTheme.typography.titleSmall,
+                    color = MaterialTheme.colorScheme.background,
+                )
+            }
+            Spacer(modifier = Modifier.weight(1f))
+            labelText?.let {
+                Text(
+                    text = labelText,
+                    style = MaterialTheme.typography.labelLarge,
+                    color = MaterialTheme.colorScheme.tertiary,
+                )
+            }
+        }
+        Spacer(modifier = Modifier.height(12.dp))
+        Text(
+            text = reason,
+            style = MaterialTheme.typography.bodyMedium,
+            color = MaterialTheme.colorScheme.onBackground,
+        )
+
+        if (askStatus != AskStatus.REJECTED) { // 신청이 승인, 대기중일 때
+            Spacer(modifier = Modifier.height(12.dp))
+            Divider(color = MaterialTheme.colorScheme.secondary)
+            Spacer(modifier = Modifier.height(12.dp))
+            Column {
+                Row {
+                    Column(
+                        modifier = Modifier
+                            .weight(1f)
+                            .padding(end = 16.dp)
+                            .align(Alignment.Bottom),
+                    ) {
+                        Row(verticalAlignment = Alignment.Bottom) {
+                            Text(
+                                text = currentLeftTime,
+                                style = MaterialTheme.typography.titleSmall,
+                                color = MaterialTheme.colorScheme.onBackground,
+                            )
+                            if (currentLeftTime.isNotBlank()) {
+                                Text(
+                                    text = " 남음",
+                                    style = MaterialTheme.typography.labelLarge,
+                                    color = MaterialTheme.colorScheme.tertiary,
+                                )
+                            }
+                        }
+                        Spacer(modifier = Modifier.height(8.dp))
+                        DodamLinearProgress(progress = progress, color = statusColor)
+                    }
+                    Column {
+                        Row(verticalAlignment = Alignment.CenterVertically) {
```

Comment:
You can reduce the overlapping code by making the code wrapped around this Row into a composable function